### PR TITLE
64tass: new recipe

### PR DESCRIPTION
**Pull request by theclue**
64tass is cross assembler targeting the 65xx series of micro processors which features the familiar syntax of Omicron TASS and TASM, one of the most popular assembler on C64 during the 80s and 90s.

Port is working perfectly, but has two small defects:

- Licensing is a bit confused. Main package is LGPL, but an embedded component has a custom one (provided) which I'm not sure is compatible with LGPL itself. No real names are provided (but demo scener/crew nicknames)

- In the INSTALL() block I copy the syntax file to vim settings dir. A better version check should be made here to proof which version of vim is installed. Have I to use a post install script?

### CODE REVIEW

File: dev-lang/64tass/64tass-1.55.recipe [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+	"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	"
+BUILD_PREREQUIRES="
+	cmd:gcc$secondaryArchSuffix
+	cmd:gzip
+	cmd:make
+	cmd:svnversion
```
Comment (korli) [inline dev-lang/64tass/64tass-1.55.recipe:40]: svnversion shouldn't be needed.

Comment (theclue) [reply dev-lang/64tass/64tass-1.55.recipe:40]: If i understood correctly, I think it is instead. It is used to check for revision number inside the makefile. It has no effect actually (let alone I get the source from a git mirror) but will break the build if absent.

Comment (korli) [reply dev-lang/64tass/64tass-1.55.recipe:40]: The makefile has to be fixed then.

File: dev-lang/64tass/64tass-1.55.recipe [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+}
+
+INSTALL()
+{
+	make install $jobArgs
+
+	# vim syntax
+	# TODO: better version checking, still ok until
+	# we are on the 8.2 branch or we repackage vim itself
+	mkdir -p $dataDir/vim/vim82/syntax
```
Comment (bitigchi) [inline dev-lang/64tass/64tass-1.55.recipe:65]: For user syntax files, user runtime configuration should be used. Vim data directory is not supposed to be user modified. In this case correct path is `~/config/settings/vim/syntax`.

Comment (pulkomandy) [reply dev-lang/64tass/64tass-1.55.recipe:65]: This is not "user", it's a system-packaged files. This is how they are handled in Haikuports and in other distros (for example youcompleteme installs itself this way).

Comment (theclue) [reply dev-lang/64tass/64tass-1.55.recipe:65]: The intense debate on the official forum on [this thread](https://discuss.haiku-os.org/t/where-to-put-vim-and-other-editors-syntax-files/9595) made clear that there is no a common way on how to handle this aside that repackaging vim itself (which is partly resolutive anyway).

**Comment by Begasus** [comment]
Any news on this?

**Comment by theclue** [comment]
> Any news on this?

At the end of the day, there's no common agreement on how to handle this. Thus, I think that unless someone is going to repackage vim (and there's no agreement on this, too), this way is the best way to package the syntax file (and it's the way other packages do). IMHO, it could be merged

**Comment by Begasus** [comment]
OK? If there is something in need of change for this it will show up I guess :)

**Pull request merged**